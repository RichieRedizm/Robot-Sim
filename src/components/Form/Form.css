/* form input styling  */
input[type='textarea'] {
  padding: 2px 10px;
  font-size: 1em;
  width: 20vw;
}
