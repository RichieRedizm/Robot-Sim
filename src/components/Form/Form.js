import React, { Fragment, useContext, useState } from 'react'
import SimContext from '../../context/sim/simContext'

const Form = () => {
  const simContext = useContext(SimContext)
  const { processCommand } = simContext
  const [command, setCommand] = useState('')

  // handles key characters entered to input and sets as state
  const handleOnChange = (e) => setCommand(e.target.value)

  // handles key down events.
  const handleOnKeyDown = (e) => {
    if (e.keyCode == 13) {
      // If key pressed is 'enter' call processCommand function
      e.preventDefault()
      processCommand(e.target.value)
    }
  }

  // handles leaving the input field and calls processCommand function
  const handleOnBlur = (e) => {
    processCommand(e.target.value)
  }

  return (
    <Fragment>
      <header>
        <title>
          <h2>Enter your command</h2>
        </title>
      </header>
      <form className='form' onKeyDown={(e) => handleOnKeyDown(e)}>
        <input
          type='textarea'
          value={command}
          placeholder='Enter Command Here....'
          onChange={(e) => handleOnChange(e)}
          onBlur={(e) => handleOnBlur(e)}
        />
      </form>
    </Fragment>
  )
}

export default Form
