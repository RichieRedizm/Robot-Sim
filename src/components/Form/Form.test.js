import { cleanup, fireEvent, getByTestId, render } from '@testing-library/react'
import React from 'react'
import SimContext from '../../context/sim/simContext'
import Form from './Form'

afterEach(cleanup)

const state = {
  position: {
    x: 0,
    y: 0,
  },
  commands: ['PLACE', 'MOVE', 'LEFT', 'RIGHT', 'REPORT'],
  directions: ['NORTH', 'SOUTH', 'EAST', 'WEST'],
  facing: 'NORTH',
  alertInfo: null,
  handleCommand: jest.fn(),
}

const renderFormComponent = (state) =>
  render(
    <SimContext.Provider value={state}>
      <Form />
    </SimContext.Provider>
  )

describe('Form component and command functionality tests', () => {
  test('Loads form component and input field plus match snapshot', () => {
    const { container, asFragment } = renderFormComponent({ ...state })
    const commandForm = getByTestId(container, 'commandForm')
    const commandInput = getByTestId(container, 'commandInput')

    expect(commandForm)
    expect(commandInput)
    expect(asFragment()).toMatchSnapshot()
  })
  test('input field accepts text update commands', () => {
    const { container } = renderFormComponent({ ...state })
    const commandInput = getByTestId(container, 'commandInput')
    expect(commandInput.value).toBe('')
    fireEvent.change(commandInput, { target: { value: 'PLACE' } })
    expect(commandInput.value).toBe('PLACE')
    fireEvent.change(commandInput, { target: { value: 'MOVE' } })
    expect(commandInput.value).toBe('MOVE')
    fireEvent.change(commandInput, { target: { value: 'LEFT' } })
    expect(commandInput.value).toBe('LEFT')
    fireEvent.change(commandInput, { target: { value: 'RIGHT' } })
    expect(commandInput.value).toBe('RIGHT')
    fireEvent.change(commandInput, { target: { value: 'REPORT' } })
    expect(commandInput.value).toBe('REPORT')
  })
})
