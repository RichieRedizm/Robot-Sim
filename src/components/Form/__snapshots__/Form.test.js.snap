// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component and command functionality tests Loads form component and input field plus match snapshot 1`] = `
<DocumentFragment>
  <header>
    <h2>
      Type your command and press enter
    </h2>
  </header>
  <form
    class="form"
    data-testid="commandForm"
  >
    <input
      data-testid="commandInput"
      placeholder="Enter Command Here...."
      type="textarea"
      value=""
    />
  </form>
</DocumentFragment>
`;
